### ISSUE TITLE
oktaSignOnPolicyRuleSignonSessionActions.go omitting MaxSessionLifetimeMinutes

### ISSUE THREAD

**Issue by eatplaysleep** [okta/okta-sdk-golang#177]
Although the Okta Developer documentation does not indicate that `maxSessionLifetimeMinutes` is an *optional* field the API will actually reject any request that does not contain it. 

Making a call without having it present with at least a value of `0` results in the following error:

```json
{
    "errorCode": "E0000001",
    "errorSummary": "Api validation failed: actions.signon.session.maxSessionLifetimeMinutes",
    "errorLink": "E0000001",
    "errorId": "oaec-6iwlpTT5S5NzaP57rESg",
    "errorCauses": [
        {
            "errorSummary": "actions.signon.session.maxSessionLifetimeMinutes: The field cannot be left blank"
        },
        {
            "errorSummary": "actions.signon.session.maxSessionLifetimeMinutes: Field must be at least 0"
        },
        {
            "errorSummary": "actions.signon.session.maxSessionLifetimeMinutes: Field can be at most 259,200"
        }
    ]
}
```

The following code needs to be changed to remove `omitempty` from `MaxSessionLifetimeMinutes`:
```go
type OktaSignOnPolicyRuleSignonSessionActions struct {
	MaxSessionIdleMinutes     int64 `json:"maxSessionIdleMinutes,omitempty"`
	MaxSessionLifetimeMinutes int64 `json:"maxSessionLifetimeMinutes,omitempty"`
	UsePersistentCookie       *bool `json:"usePersistentCookie,omitempty"`
}
```

**Comment by bretterer** [comment]
@eatplaysleep thank you for the report here!  I hear your concerns with `omitempty`, however as part of our generation of the SDK, we place `omitempty` here as a safeguard against a flexible API that may or may not include all properties when returned from the API calls.  

Are you getting this error when making a call to the API, or as part of the return from the API?

**Comment by eatplaysleep** [comment]
Makes sense. I receive the error when _making_ the call. Both from Terraform and also just a Postman request.

**Comment by bretterer** [comment]
@eatplaysleep thanks for the information.  So it sounds like this is a documentation confusion on our part because of our use of `omitempty`.  It seems you were assuming this property was optional because of the `omitempty`, is that correct?  If thats the case, outside of removing the `omitempty` what would have made it clear for you so we can update that documentation to help others?

**Comment by eatplaysleep** [comment]
I understand keeping the `omitempty`. I probably should have omitted that from the ticket in the first place and just focused on the issue. :) 

I suppose it is a matter of either updating the API so that the field is actually optional _or_ at least updating the developer documentation to indicate it is required. 

I'm running into this issue using the Okta Terraform provider (which would also need a doc update) so I just had to set a default value of `0` for this particular field instead of `null` in order to work around it.

**Comment by bretterer** [comment]
@eatplaysleep thank you for that information.  I will pass this along to our team to figure out out next steps.

**Comment by bogdanprodan-okta** [comment]
This issue is fixed in the latest v2.2.2
